internal/kongstate/kongstate.go

```diff
@@ -87,6 +88,19 @@ func (ks *KongState) FillConsumersAndCredentials(log logrus.FieldLogger, s store
 					credConfig[k] = strings.Split(string(v), ",")
 					continue
 				}
+				// TODO this is a kongCredType-agnostic mutation that should only apply to Oauth2 credentials.
```

Comment:
ideally a TODO should be accompanied by an issue link.

Reply:
Ideally yeah, but I'm not really sure how to describe it other than "the cred system is a pain to work with, make it better".